engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigrationState.java

```diff
@@ -347,7 +347,7 @@ public void migrateElementInstancePopulateProcessInstanceByDefinitionKey() {
           if (recordValue.getBpmnElementType() == BpmnElementType.PROCESS) {
             elementInstanceKey.wrapLong(key.getValue());
             processDefinitionKey.wrapLong(recordValue.getProcessDefinitionKey());
-            processInstanceKeyByProcessDefinitionKeyColumnFamily.insert(
+            processInstanceKeyByProcessDefinitionKeyColumnFamily.upsert(
```

Comment:
❓ Interesting catch. So that I understand correctly. This is needed because followers build state alongside the leader without having ran the migration tasks. When they become leader the migration is run, in which case there may already be data in this column family.

Reply:
No, you will hate me for this. It is because the Event Appliers changed behaviour 😱 

Before when we inserted an element instance it would not put data in this new ColumnFamily. Now when we insert an element instance it does insert data in the new ColumnFamily.
This means that when events are replayed it results in entries already being part of this ColumnFamily. When we migrate it could result in duplicate keys, hence the need for upsert.

I figured this was more pragmatic than creating a 2nd version of the ElementActivatingApplier and having a second version of the DbElementInstanceState so that creating a new instance doesn't insert into this ColumnFamily on replay 🙃

Reply:
Although your comment might be valid as well 🤔